Added footprint for LED_RGB_Everlight_EASV3015RGBA0

fabikroeger opened the pull request:
Added footprint for LED_RGB_Everlight_EASV3015RGBA0
Datasheet: https://everlightamericas.com/index.php?controller=attachment&id_attachment=3220
PR symbol: https://github.com/KiCad/kicad-symbols/pull/2242
PR 3d model:
Screenshot:
![Everlight_LED_rightangeled](https://user-images.githubusercontent.com/6748040/66484240-e4773880-eaa6-11e9-97ec-96b419d4f283.png)
![Everlight_LED_rightangeled_footprint](https://user-images.githubusercontent.com/6748040/66484270-f22cbe00-eaa6-11e9-8472-3738b742ca53.png)


---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the footprint(s) you are contributing
- [x] An example screenshot image is very helpful 
- [x] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
- [x] Give a reason behind any intentional library convention rule violation.

gmsotavio commented:
Hi, and thanks for your first contribution! I'm not reviewing this footprint at this time. It's just my first thoughts about it.

>Travis is upset because of F9.3 is violated but the 3d model is proper aligned.

Rotation and offset must be 0 (The model must be aligned correctly to the footprint within the 3d modeling software.). See footprint 3D model requirements at (http://kicad-pcb.org/libraries/klc/F9.3/).
![image](https://user-images.githubusercontent.com/10669626/66693208-ed216780-ec7c-11e9-929f-b2c79b7946bb.png)

- [ ] Courtyard clearance must be 0.25mm. The courtyard of your footprint is overlapping fabrication layer elements. See courtyard layer requirements at (http://kicad-pcb.org/libraries/klc/F5.3/)
![image](https://user-images.githubusercontent.com/10669626/66693260-720c8100-ec7d-11e9-910d-8f87764abf30.png)

- [ ] The location of the anchor seems wrong according to the datasheet. See footprint anchor placement at (http://kicad-pcb.org/libraries/klc/F6.2/). Take a special look at **Exceptions** section.

![image](https://user-images.githubusercontent.com/10669626/66700612-d1dc4980-ecc8-11e9-84cd-9f125c7af8a3.png)

- [ ] Please, provide a better description. Only the datasheet link is not enough.

![image](https://user-images.githubusercontent.com/10669626/66700797-795a7b80-eccb-11e9-8184-7928d8db1095.png)

Provide the fixes, and I will review your contribution in-depth,

gmsotavio commented:
- [ ]  Use only one arc.
![image](https://user-images.githubusercontent.com/10669626/66716785-40dc9f80-eda8-11e9-8a92-51ce71ecee28.png)

- [ ] The Fab layer must represent the nominal dimensions of the real part. Please, change your Fab outline as depicted below.
![image](https://user-images.githubusercontent.com/10669626/66716906-91083180-eda9-11e9-8ba7-5f8ac0c340a1.png)

- [ ] Except by footprint pads, most dimensions are fully wrong, including Fab outline and Courtyard clearance. Take a look in the partial dimensioned drawing depicted bellow.
![image](https://user-images.githubusercontent.com/10669626/66717096-a1b9a700-edab-11e9-809f-6e1e632a6acc.png)

- [ ] Pad rotation must be 0 degrees
![image](https://user-images.githubusercontent.com/10669626/66719104-d6d0f400-edc1-11e9-9b18-3de2b401b004.png)

- [ ] The datasheet is very unclear about sizing the component and also about its position in the footprint. I took some measurements from the 3D step model and made some assumptions based on the datasheet. Take a look at the following drawing. Try to following it. You have placed the pick and place point in a random place. Take care of the right pick and place location. For manual soldering, I do not see any problem, but for automated assembly, this can cause serious problems.
![image](https://user-images.githubusercontent.com/10669626/66716712-4a193c80-eda7-11e9-9d1c-61c4ab67c17a.png)

Please, check carefully your contribution before request a new review round. The issues found are acceptable for a first-time contributor.

poeschlr commented:
It might also be a good idea to get the silk outline closer fitting but have the pads (plus some clearance) removed from it. This would mean the outline is more useful for hand assembly.

fabikroeger commented:
Hi gmsotavio,

thankyou very much for that in-depth review. I fixed those remarks and commented one below related to that pick and place anchor which is really wanky documented.


> * [x]   Use only one arc.
>   ![image](https://user-images.githubusercontent.com/10669626/66716785-40dc9f80-eda8-11e9-8a92-51ce71ecee28.png)
Fixed.
> * [x]  The Fab layer must represent the nominal dimensions of the real part. Please, change your Fab outline as depicted below.
I have read again the datasheet and came then to the Fab Layer dimensions in the actual PR. 
>   ![image](https://user-images.githubusercontent.com/10669626/66716906-91083180-eda9-11e9-8ba7-5f8ac0c340a1.png)
> * [x]  Except by footprint pads, most dimensions are fully wrong, including Fab outline and Courtyard clearance. Take a look at the partial dimensioned drawing depicted bellow.
>   ![image](https://user-images.githubusercontent.com/10669626/66717096-a1b9a700-edab-11e9-809f-6e1e632a6acc.png)
Here also... i have read the datasheet again and also the KLC rules for courtyard and fab layer.
> * [x]  Pad rotation is 180 deg, but must be 0 deg
>   ![image](https://user-images.githubusercontent.com/10669626/66719104-d6d0f400-edc1-11e9-9b18-3de2b401b004.png)
> * [x]  The datasheet is very unclear about sizing the component and also about its position in the footprint. I took some measurements from the 3D step model and made some assumptions based on the datasheet. Take a look at the following drawing (click to increase the size). Try to following it. You have placed the pick and place point in a random place. Take care of the right pick and place location. For manual soldering, I do not see any problem, but for automated assembly, this can cause serious problems.

>   ![image](https://user-images.githubusercontent.com/10669626/66719266-008b1a80-edc4-11e9-932e-25f2aca30f33.png)

> 
> Note: Don't worry about F6.2. The script works well for symmetrical parts but fails for nonsymetrical. For now, you can ignore Travis CI.
> 
> Please, check carefully your contribution before request a new review round. The issues found are acceptable for a first-time contributor.

One more thing: I used again the step file format to hold the 3d data. I had scaling problems with the export of wrl files from freecad.

gmsotavio commented:
Hi @fabikroeger. Sorry by the delayed response. I have checked your changes but I need more some days to give a proper return to you, with a dimensioned drawing. I would like to see this footprint merged as soon. Again, I apologize for the delay.

fabikroeger commented:
Hi @gmsotavio,   
thank you for your effort so far! I am in vacation until end of november. So do't wonder if a am not reacting.

fabikroeger commented:
Bump :-)

poeschlr commented:
The silk reference and the value field need to exchange position (reference is always to the "north")
Do not reduce the size of the value field (must be 1mm)

3d path must point to the wrl file (kicad replaces it with step where required)
Do not use "right-angle" in the description or tags, use horizontal.

I would also really prefer the silk to be much closer. Something like this:
![Screenshot from 2020-02-04 23-10-49](https://user-images.githubusercontent.com/18327350/73792237-2ccc0c00-47a4-11ea-8b24-a8e8131fc010.png)

---

There are also errors in the dimensions
![Screenshot from 2020-02-04 23-39-50](https://user-images.githubusercontent.com/18327350/73793948-b3ceb380-47a7-11ea-864b-011202557a42.png)

myfreescalewebpage commented:
@fabikroeger are you still alive on this subject ? If yes can you follow the changes required above by @poeschlr

myfreescalewebpage commented:
No news of the author, indicate this is Abandoned
